src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs

```diff
@@ -38,6 +38,9 @@ public RegionInfo(string region, RegionAutodetectionSource regionSource, string
         private readonly IHttpManager _httpManager;
         private readonly int _imdsCallTimeoutMs;
 
+        // lock for discovery. So it is done only once
+        private readonly SemaphoreSlim _lockDiscover = new SemaphoreSlim(1);
```

Comment:
We create 1 RegionManager per Application object, so 2 different apps would have 2 different locks. Notice how we use statics for autoDiscoveredRegion etc. Consider making this lock static as well.